downsample_rs/src/helpers.rs

```diff
@@ -0,0 +1,57 @@
+#[cfg(feature = "half")]
+use half::f16;
+use ndarray::ArrayView1;
+
+// ------------ AVERAGE
+
+// TODO: future work -> this can be optimized by using SIMD instructions (similar to the argminmax crate)
+// TODO: this implementation can overfow
```

Comment:
Will leave this as is (for now), existing LTTB implementations also suffer from this & no issue is raised..
(I even believe the mean implementation in numpy is susceptible to this)

Reply:
See the `umr` -> https://github.com/numpy/numpy/blob/8cec82012694571156e8d7696307c848a7603b4e/numpy/core/_methods.py#L164